conda_build/index.py

```diff
@@ -937,7 +939,7 @@ def index_subdir(self, subdir, index_file=None, verbose=False, progress=False):
             new_repodata_packages = {k: v for k, v in old_repodata.get('packages', {}).items() if k in unchanged_set}
             new_repodata_conda_packages = {k: v for k, v in old_repodata.get('packages.conda', {}).items() if k in unchanged_set}
```

Comment:
We're using these two dicts for set operations below.  It might be worth our memory to create a new set using their keys and use that for the set operations.

```python
new_repodata_packages = {k: v for k, v in old_repodata.get('packages', {}).items() if k in unchanged_set}
new_repodata_conda_packages = {k: v for k, v in old_repodata.get('packages.conda', {}).items() if k in unchanged_set}
repodata_packages = set(new_repodata_packages).union(new_repodata_conda_packages)
```

Then, later, do this:
```python
if k not in repodata_packages:
```

Reply:
Uh, nevermind - I see this is a cheap operation.